File: frontend/src/types/reports.ts
export interface DashboardSavingGoal {
  name: string;
  progress_percentage: number;
}

export interface DashboardTransaction {
  id: string;
  description: string;
  amount: number;
  type: "Income" | "Expense" | "Transfer";
  date: string;
}

export interface DashboardBudget {
  category_name: string;
  planned: number;
  actual: number;
  remaining: number;
}

export interface DashboardData {
  total_balance: number;
  income_this_month: number;
  expense_this_month: number;
  net_balance: number;
  recent_transactions: DashboardTransaction[];
  saving_goals: DashboardSavingGoal[];
  budget_summary: DashboardBudget[];
}

File: backend/app/schemas/reports.py
import uuid
from datetime import date
from typing import List, Optional
from pydantic import BaseModel

class DashboardSavingGoal(BaseModel):
    name: str
    progress_percentage: float

class DashboardTransaction(BaseModel):
    id: uuid.UUID
    description: str
    amount: float
    type: str  # Income, Expense, Transfer
    date: date

class DashboardBudget(BaseModel):
    category_name: str
    planned: float
    actual: float
    remaining: float

class DashboardData(BaseModel):
    total_balance: float
    income_this_month: float
    expense_this_month: float
    net_balance: float
    recent_transactions: List[DashboardTransaction]
    saving_goals: List[DashboardSavingGoal]
    budget_summary: List[DashboardBudget]

class DashboardEnvelopeResponse(BaseModel):
    success: bool = True
    message: str = "Dashboard metrics loaded"
    data: DashboardData
